src/widgets/block.rs

```diff
@@ -396,16 +400,38 @@ impl<'a> Block<'a> {
         self
     }
 
-    /// Defines the block style.
+    /// Defines the style of the entire block.
     ///
     /// This is the most generic [`Style`] a block can receive, it will be merged with any other
-    /// more specific style. Elements can be styled further with [`Block::title_style`] and
-    /// [`Block::border_style`].
+    /// more specific styles. Elements can be styled further with [`Block::title_style`] and
+    /// [`Block::border_style`], which will be applied on top of this style. If the block is used
+    /// as a container for another widget, the inner widget can also be styled.
     ///
     /// `style` accepts any type that is convertible to [`Style`] (e.g. [`Style`], [`Color`], or
     /// your own type that implements [`Into<Style>`]).
     ///
-    /// This will also apply to the widget inside that block, unless the inner widget is styled.
+    /// This will also apply to the widget inside that block, the inner widget overrides the style.
```

Comment:
At some point I think we should rephrase this? This says "inside that block" and "inner widget" but that isn't how the code works and I don't think we use that language on the website docs.

Reply:
Can you suggest alternate wording for this?

Reply:
The shortest change I can think of is:

```suggestion
    /// This will also apply to the widget that houses the block, in which case the widget's styles will be applied on top of the block's styles.
```

But I'm fine with scoping this PR to not include this kind of change because I think we'd have to read through all the docs with an eye to make them consistent.

Reply:
Hmm, isn't the original docs and the change in this PR (and my previous comment suggestion) all incorrect?

For a `Paragraph`:

https://github.com/ratatui-org/ratatui/blob/cbfcd34ef5cfaa28f72706af373b7dfe4a648795/src/widgets/paragraph.rs#L332-L334

The code for a `Paragraph` applies the `block` style _after_ applying the paragraph's own style. I always thought it was the other way around. I haven't checked all the widgets for consistency here.

Reply:
The `block`'s style overlaps or is "painted over" the paragraph's style. In the actual behavior, the block's style "overrides" the paragraph's style. 

I also think "overrride" implies that the block's style replaces the paragraph's style but really the block's style is superimposed on top of the paragraph's style. A paragraph style with background white and a block style with foreground red will result in "red on white" behavior, right?

Reply:
This comment refers to widgets that are rendered in the Block::inner rect, not in widgets that have blocks.

> The code for a Paragraph applies the block style after applying the paragraph's own style. I always thought it was the other way around. I haven't checked all the widgets for consistency here.

This is the consistent behavior for all widgets that have a block and widget style. If this was the other way around, then styling borders would be impossible because the styles set on the widget would override the block. I have a vague recollection this is covered in an issue / PR comment from eons ago, but the behavior is correct either way.

> I also think "overrride" implies that the block's style replaces the paragraph's style but really the block's style is superimposed on top of the paragraph's style. A paragraph style with background white and a block style with foreground red will result in "red on white" behavior, right?

Override here means the same as what we're taking about in terms of merging in the style docs:

> Styles represents an incremental change. If you apply the styles S1, S2, S3 to a cell of the terminal buffer, the style of this cell will be the result of the merge of S1, S2 and S3, not just S3.

The problem with the word merge is that it doesn't imply a precedence of the items that are merged, whereas override does imply this.

Reply:
> This comment refers to widgets that are rendered in the Block::inner rect, not in widgets that have blocks.

My bad, I don't think I caught that it is referring to `Block::inner`. I still don't quite follow what this line is supposed to mean then. The following line was just above it, so I read it as widgets that have a block.

```rust
/// If the block is used as a container for another widget, the inner widget can also be styled.
```

> The problem with the word merge is that it doesn't imply a precedence of the items that are merged, whereas override does imply this.

Maybe "overriding merge" would solve my nit here.

Reply:
None of my comments are blocking this PR!

Reply:
tweaked it a little 05389d5b8